### DISCUSSION TITLE
V3 beta24 errors

### DISCUSSION THREAD

**Discussion by jcgmalave** [filamentphp/filament#7331] [Help]
i get the following errors during the latest v3 upgrade:

1) Typed property Filament\Widgets\TableWidget::$table must not be accessed before initialization

2) Class "NumberFormatter" not found

3)
[vite:css] Failed to find '../../vendor/filament/filament/resources/css/app.css'
  in [
    C:\xampp\htdocs\modatex-chile-three\resources\css
  ]
file: C:\xampp\htdocs\modatex-chile-three\resources\css\filament.css

Any tips? Thanks

**Comment by leandrocfe** [comment]
What about beta26?
Can you run `composer update` and try again?

**Comment by sumardi** [comment]
Please make sure the `extension=intl` is enabled in php.ini

**Comment by wivaku** [comment]
Same: getting `Typed property Filament\Widgets\TableWidget::$table must not be accessed before initialization`.

Using beta27, with intl enabled.

Also: TableWidget is not mentioned in v3 documentation.
v3: https://beta.filamentphp.com/docs/3.x/widgets/installation/ (only stats and charts)
v2: https://filamentphp.com/docs/2.x/admin/dashboard/tables

**Reply by sumardi** [reply]
I believe TableWidget is still a thing in v3. Share stack trace.

**Reply by wivaku** [reply]
TableWidget is definitely [still a thing in v3](https://github.com/filamentphp/filament/blob/3.x/packages/widgets/src/TableWidget.php). But it's missing from the v3 documentation. Stack trace: https://flareapp.io/share/pPvQEw9m

**Reply by danharrin** [reply]
Thanks for reporting, I fixed it

**Comment by sumardi** [comment]
> TableWidget is definitely [still a thing in v3](https://github.com/filamentphp/filament/blob/3.x/packages/widgets/src/TableWidget.php). But it's missing from the v3 documentation. Stack trace: https://flareapp.io/share/pPvQEw9m

Use the `table()` method to configure the table widget.

```php
public function table(Table $table): Table
    {
        return $table
            ->query(\App\Models\Order::where("status_id", 2)->orderBy('id', 'desc'))
            ->columns([
                Tables\Columns\TextColumn::make('id')->label('#'),
                Tables\Columns\TextColumn::make('full_name')->label(__('Name')),
            ])
            ->filters([
                // ...
            ])
            ->actions([
                // ...
            ])
            ->bulkActions([
                // ...
            ]);
    }
```

**Reply by wivaku** [reply]
that was it, thanks! 
I was still using the v2 definition (`protected function getTableColumns(): array`). Did not realise that changed in recent beta.

**Comment by alkoumi** [comment]
I am facing this issue now and I found the solution is to Disable the lazy loading of the TableWidget

```
protected static bool $isLazy = false;
```

**Reply by Sparrowan** [reply]
Thanks this works well.

**Reply by alkoumi** [reply]
Glad to helping any filament lover

**Reply by arnaudliperini** [reply]
Still happening in v4. Thanks this solved it.
